### ISSUE TITLE
Errors when trying to run LightUpPi Alarm

### ISSUE THREAD

**Issue by a61c** [carlosperate/LightUpPi-Alarm#1]
I have been trying to set up LightUpPi and have been getting these errors everytime I try to run it. I have all the dependencies that are listed here installed.

![dqlja6lg](https://cloud.githubusercontent.com/assets/18472707/14546106/f091f0d0-029a-11e6-9d23-5699ad551bf4.png)

**Comment by carlosperate** [comment]
Looks like this particular iterator from dataset has changed quite a few times. It took me some digging around, but if you install version 0.5.4 it should work, uninstall your current version and then:

```
pip install -I dataset==0.5.4
```

**Comment by carlosperate** [comment]
I've updated the requirements.txt and README instructions to indicate this.

Let me know if it works for you and thanks for reporting it!

**Comment by a61c** [comment]
That's great - it now works perfectly. Thanks for sorting this out so
quick!

Just one other problem now, the CLI works great, however when I run the
server and try to use the web interface it gives an error about being
unable to find the alarms, so it defaults to demo alarms - But when I look
at my alarms through the CLI, it shows my custom alarms just fine.

Thanks again.

**Comment by carlosperate** [comment]
did you try using the "-b" command for having both the command line and web interface?

**Comment by a61c** [comment]
I have, it still gives the same error :-(

**Comment by carlosperate** [comment]
when you run with only the server, does it print anything on the console?

**Comment by a61c** [comment]
No errors,
Here's a screenshot...
http://imgur.com/E9NHtlC

**Comment by carlosperate** [comment]
and when connecting to http://localhost:80 it doesn't print anything on the console?

**Comment by a61c** [comment]
I am getting the Get commands when opening the page from another device:
http://imgur.com/bQpUnLd

**Comment by carlosperate** [comment]
so it works when accessed from a different device? but not when executed from the pi?

**Comment by a61c** [comment]
Sorry, I should have been clearer, the page loads OK from another device or on the pi, however it doesn't show or let me set my own alarms.
http://imgur.com/2Rw2X6

Do you have an apk for the LightUpDroid Alarm app, then I could quickly give that a try?

**Comment by a61c** [comment]
Sorry incorrect link, should be...
http://imgur.com/2Rw2X6l

**Comment by carlosperate** [comment]
Yeah, adding an alarm from the web interface doesn't work for me either. I suspect I never finished that feature, as I used the Android app pretty much exclusively once I had that running, but I honestly cannot remember, it's been a while.

I don't really have a copy of the Android apk on my main computer, I'll have a look in my old laptop once I get a chance, but if there isn't a copy there either you would have to build from source.

**Comment by a61c** [comment]
No problem, thanks for your help. I'll have a go at building the apk, when I get chance. Unfortunately the alarms don't even display on the web interface.

Thanks anyway.
